pkg/generic/thrift/http_go116plus_amd64.go

```diff
@@ -0,0 +1,68 @@
+//go:build amd64 && go1.16
+// +build amd64,go1.16
+
+/*
+ * Copyright 2021 CloudWeGo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package thrift
+
+import (
+	"context"
+
+	"github.com/apache/thrift/lib/go/thrift"
+	"github.com/cloudwego/dynamicgo/conv"
+	"github.com/cloudwego/dynamicgo/conv/j2t"
+
+	"github.com/cloudwego/kitex/pkg/generic/descriptor"
+	"github.com/cloudwego/kitex/pkg/protocol/bthrift"
+	"github.com/cloudwego/kitex/pkg/remote/codec/perrors"
+	cthrift "github.com/cloudwego/kitex/pkg/remote/codec/thrift"
+)
+
+// Write ...
+func (w *WriteHTTPRequest) Write(ctx context.Context, out thrift.TProtocol, msg interface{}, requestBase *Base) error {
+	if w.enableDynamicgo {
+		req := msg.(*descriptor.HTTPRequest)
+		if w.svc.DynamicgoDsc == nil {
+			return perrors.NewProtocolErrorWithMsg("svcDsc.DynamicgoDsc is nil")
+		}
+		tyDsc := w.svc.DynamicgoDsc.Functions()[w.method].Request().Struct().Fields()[0].Type()
+
+		body := req.GetBody()
+		buf := make([]byte, fieldBeginLen, len(body)+structWrapLen)
+		offset := 0
+		offset += bthrift.Binary.WriteStructBegin(buf[offset:], "")
+		bthrift.Binary.WriteFieldBegin(buf[offset:], "", thrift.STRUCT, 1)
+		ctx = context.WithValue(ctx, conv.CtxKeyHTTPRequest, req)
+		cv := j2t.NewBinaryConv(w.opts)
+		if err := cv.DoInto(ctx, tyDsc, body, &buf); err != nil {
```

Comment:
The input param is &buf not &buf[offset:]?

Reply:
@jayantxie DoInto() appends the data, so it's possible just to pass &buf.

Reply:
I mean, since you've written the StructBegin and FiledBegin to the buf, how can you not modify the offset of it and directly pass it into dynamicgo?

Reply:
Does DynamicGo know the first 8 bytes are StructBegin and FiledBegin which you've written?

Reply:
@jayantxie Dynamicgo just **appends** (not **overwrites**) new data to buf, for example, assuming that value passed to dynamicgo is [1,2,3] and the thrift data after conversion by dynamicgo is [4,5,6], the buf returned by dynamicgo will be [1,2,3,4,5,6].
So I think it does not really matter to dynamicgo what is already written in buf, and it does not matter whether we pass &buf or &buf[offset:] to dynamicgo.